viewing_party/main.py

```diff
@@ -0,0 +1,182 @@
+# main.py
+
+# --------
+# WAVE ONE
+# --------
+
+def create_movie(movie_title, genre, rating):
+    """
+    Creates and returns a movie dictionary. Dictionary contains information
+    about the movie.
+    """
+    if movie_title == None or genre == None or rating == None:
+        return None
+    movie = {}
+    movie['title'] = movie_title
+    movie['genre'] = genre
+    movie['rating'] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    """
+    Adds a movie to the user's watched list.
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """
+    Adds a movie to the user's watchlist.
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    """
+    Moves movie from user's watchlist to watched list.
+    """
+    for movie in user_data['watchlist']:
+        if movie['title'] == movie_title:
+            user_data['watched'].append(movie)
+            user_data['watchlist'].remove(movie)
+            return user_data
+    else:
+        return user_data
+
+
+# --------
+# WAVE TWO
+# --------
+
+def get_watched_avg_rating(user_data):
+    """
+    Returns the average rating of the movies the user has watched. Works by summing
+    the ratings, then dividing by number of movies.
+    """
+    if len(user_data['watched']) > 0:
+        sum = 0
+        for movie in user_data['watched']:
+            sum += movie['rating']
+        return sum / len(user_data['watched'])
+    else:
+        return 0
+
+def get_most_watched_genre(user_data):
+    """
+    Returns the most-watched genre in the user's data, or None if no movies in watched
+    list. Works by looping through the user data to compile a dictionary of genres the 
+    user has watched and how many times each genre has been watched. Determines the most-
+    watched genre by finding the maximum value in the dictionary.
+    """
+    genres = {}
+
+    if len(user_data['watched']) == 0:
+        return None
+
+    for movie in user_data['watched']:
+        if movie['genre'] in genres:
```

Comment:
Consider storing movie["genre"] in a variable and then using it in the rest of the code in this section. How does that change the readability of the lines that follow?